src/Features/Core/Portable/UseIsNullCheck/AbstractUseIsNullDiagnosticAnalyzer.cs

```diff
@@ -108,9 +119,18 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context, IMethodSymbol refe
                 return;
             }
 
+            var properties = ImmutableDictionary<string, string>.Empty;
+
+            switch (GetGenericParameterConstraintType(syntaxFacts, semanticModel, arguments[0], arguments[1], cancellationToken))
```

Comment:
a simpler approach seems like it would be if you just had this method return the type parameter if there is one.

then you could do:

```c#
if (typeParameter?.HasValueConstraint == true)
{
  return;
}
else if (...?.HasReferenceConstraint == true)
{
    ...
}
```

Then there's no need for the enum to shuttle data across that's already in the ITypeParameter.